### DISCUSSION TITLE
Is GGN model implemented in GnPy to tackle C+L band systems?

### DISCUSSION THREAD

**Discussion by mukherjeearup** [Telecominfraproject/oopt-gnpy#457] [Q&A] [answered]
I can see that in "science_utils.py" file, GGN model is being implemented within "Nli solver" class in "_ggn_spectrally_separated" method (The method uses eq. 21 from arXiv: 1710.02225)
So, is this method really used in Gnpy to compute NLI specially for wide band systems? If yes, then how to get the NLI values computed from GGN model from GnPy?
Also, I can see that "_ggn_spectrally_separated" method uses SRS as an argument, so is SRS tilt already taken care within this method or do we need to calculate SRS tilt separately?
Please help to answer my questions :)

**Comment by jktjkt** [comment]
@AndreaDAmico , can you please take a look?

**Comment by AndreaDAmico** [comment, marked as answer]
The GGN model implementation in GNPy can be enabled setting the NLI parameter "method", within the simulation parameters, equal to "ggn_spectrally_separated" (the default use an implementation of the GN model). 
There are two ways to set the simulation parameters depending on how you are running GNPy:

1. **Running from command line** Using the command `gnpy-transmission-example`, you can use the option --sim <simulation parameters filename> to pass a simulation parameters JSON file where you can specify the NLI method to be used (you can find a template of the simulation parameters JSON file in gnpy/example-data/sim_params.json).

2. **Running GNPy in Python** You can import the class `Sim_Params` from the module gnpy.core.parameters, get assign the instance of the simulations parameters with `sim_params = Sim_Params.get()` and setting the NLI method as follows: `sim_params.nli_params.method = "ggn_spectrally_separated"`

In general, the NLI method used for calculating the NLI noise is decoupled from the SRS, meaning that the GNPy implementation of the NLI solver evaluated the NLI noise given a fiber loss/gain profile along the spectrum frequencies and the fiber length. The evaluation of the loss/gain profile is completely managed by the Raman solver, within GNPy, which may consider or leave out the SRS effect. In order to consider the SRS effect you have to set the Raman parameter "flag", within the simulation parameters, equal to `True` (the default would leave out the Raman effect from the calculation considering only the intrinsic fiber loss coefficient) as you do for the NLI parameter "method".

To summarized, GNPy is able to evaluate the GGN including the SRS effect on its own, you just have to set the NLI parameter "method" equal to "ggn_spectrally_separated" and the Raman parameter "flag" equal to `True` within the simulation parameters.
